src/network/core/packet.cpp

```diff
@@ -370,6 +369,22 @@ std::vector<byte> Packet::Recv_buffer()
 	return data;
 }
 
+/**
+ * Extract at most the length of the span bytes from the packet into the span.
+ * @param span The span to write the bytes to.
+ * @return The number of bytes that were actually read.
+ */
+size_t Packet::Recv_bytes(std::span<byte> span)
+{
+	auto tranfer_to_span = [](std::span<uint8_t> destination, const char *source, size_t amount) {
```

Comment:
Is the `uint8_t` versus `byte` in the function parameter intended?

Reply:
Not really.

Reply:
Didn't we want to go for uint8_t instead of byte?

Reply:
The current way keeps `Packet` consistent with respect to the 'byte' type.

It would be weird if the types of Send and Recv would differ, and I wanted to keep this as small as possible to prevent having to dig myself out of a rabbit hole once again.

Moving to `uint8_t` would be something for another PR.